src/Processors/QueryPlan/Optimizations/filterPushDown.cpp

```diff
@@ -42,11 +42,18 @@ static bool filterColumnIsNotAmongAggregatesArguments(const AggregateDescription
     return true;
 }
 
-static size_t
-tryAddNewFilterStep(QueryPlan::Node * parent_node, QueryPlan::Nodes & nodes, const Names & allowed_inputs,
-                    bool can_remove_filter = true, size_t child_idx = 0)
+static void checkChildrenSize(QueryPlan::Node * child_node, size_t child_idx )
+{
+    auto & child = child_node->step;
+    if (child_idx >= child->getInputStreams().size() || child_idx >= child_node->children.size())
```

Comment:
What if children are empty?

Reply:
Changed to pass expected child number `child_num` instead of index and added comment